dashboard/src/actions/profileActions.js

```diff
@@ -0,0 +1,92 @@
+import * as TYPES from "./types";
+import API from "../utils/axiosInstance";
+import * as API_ROUTES from "../utils/apiConstants";
+import Cookies from "js-cookie";
+
+export const getProfileDetails = () => async (dispatch) => {
+  try {
+    dispatch({ type: TYPES.LOADING });
+    const username = Cookies.get("username") ? Cookies.get("username") : null;
+
+    const response = await API.get(`${API_ROUTES.GET_USER}/${username}`);
+
+    if (response.status === 200 && Object.keys(response.data).length > 0) {
+      dispatch({
+        type: TYPES.GET_USER_DETAILS,
+        payload: response?.data,
+      });
+    } else {
+      let toast = {
+        variant: "danger",
+        title: response?.data?.message,
+      };
```

Comment:
Why do the `authActions` methods assume that API errors go on the `catch(error)` path while here the API error handling is on an `else` and the `catch` seems to assume total `HTTP` "network error" failure? Is this `API.get` vs `API.post`, or is there something else going on that I'm missing?